src/api/habits/adapters/json.ts

```diff
@@ -9,24 +9,56 @@ const jsonHabitServiceFactory = (): HabitService => {
   const addHabit = (): Promise<string> =>
     Promise.resolve('adding habit in ze factorie');
 
-  const getHabits = (): Promise<GoalWithHabitHistory[]> =>
-    fetch('../data/habits.json')
-      .then((res) => res.json() as Promise<Response>)
-      .then((json) =>
-        json.map(({ habits, ...rest }) => ({
-          ...rest,
-          habits: habits.map(({ entries, frequency, ...restHabits }) => ({
-            ...restHabits,
-            frequency: frequency as Frequency,
-            entries: entries.map(({ completionDate, ...restEntries }) => ({
-              ...restEntries,
-              completionDate: new Date(completionDate),
-            })),
-          })),
-        }))
+  const parseJsonHabits = (json: string) => {
+    const jsonHabits = JSON.parse(json) as unknown as Response;
+
+    return jsonHabits.map(({ habits, ...rest }) => ({
+      ...rest,
+      habits: habits.map(({ entries, frequency, ...restHabits }) => ({
+        ...restHabits,
+        frequency: frequency as Frequency,
+        entries: entries.map(({ completionDate, ...restEntries }) => ({
+          ...restEntries,
+          completionDate: new Date(completionDate),
+        })),
+      })),
+    }));
+  };
+
+  const getHabits = (): Promise<GoalWithHabitHistory[]> => {
+    const localHabits = localStorage.getItem('habits');
+    console.log('localHabits', localHabits);
+    if (!localHabits) return Promise.resolve([]);
+
+    try {
+      return Promise.resolve(parseJsonHabits(localHabits));
+    } catch (error) {
+      console.error('Error fetching habits');
+      return Promise.reject(new Error('Failed to fetch habits'));
+    }
+  };
+
+  const saveHabits = (habits: GoalWithHabitHistory[]) => {
+    try {
+      localStorage.setItem('habits', JSON.stringify(habits));
+    } catch (error) {
+      return Promise.reject(
+        new Error('Failed to save habits to local storage.')
       );
+    }
+
+    return Promise.resolve('Saved habits successfully.');
```

Comment:
```suggestion
    return Promise.resolve();
```
returning nothing here may be better than getting into a habit of returning "logging messages".